src/dds_manager.cpp

```diff
@@ -979,20 +1016,26 @@ bool DDSManager::replaceFilter(const std::string& topicName,
     subscriber->delete_datareader(dataReader);
     DDS::TopicDescription* targetTopic = nullptr;
 
-
     // Create a new filtered topic if requested
     if (!filter.empty())
     {
         // The topic filter name must be unique or it will fail on the
         // second time it's created
-        static int counter = 0;
+        int counter = 0;
+        if (!existingFilterName.empty())
+        {
+            std::size_t found = existingFilterName.find_last_of("_");
+            std::string count_string = existingFilterName.substr(found + 1);
+            counter = std::stoi(count_string);
+        }
         ++counter;
         const std::string filterName =
             topicName + "_" +
             readerName + "_" +
             std::to_string(counter);
 
-        const DDS::StringSeq noParams;
+        //Create the content filter with no swappable params
+        const DDS::StringSeq noParams = DDS::StringSeq();
```

Comment:
```suggestion
        const DDS::StringSeq noParams;
```
default constructor is sufficient

Reply:
Good find.